common/persistence/pinot/pinot_visibility_store_test.go

```diff
@@ -0,0 +1,1546 @@
+	request := &p.InternalListWorkflowExecutionsByTypeRequest{}
+
+	tests := map[string]struct {
+		request       *p.InternalListWorkflowExecutionsByTypeRequest
+		expectedResp  *p.InternalListWorkflowExecutionsResponse
+		expectedError error
+	}{
+		"Case1: normal case with nil response": {
+			request:       request,
+			expectedResp:  nil,
+			expectedError: nil,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockPinotClient := pnt.NewMockGenericClient(ctrl)
+			mockProducer := &mocks.KafkaProducer{}
+			mgr := NewPinotVisibilityStore(mockPinotClient, &service.Config{
+				ValidSearchAttributes:  dynamicconfig.GetMapPropertyFn(definition.GetDefaultIndexedKeys()),
+				ESIndexMaxResultWindow: dynamicconfig.GetIntPropertyFn(3),
+			}, mockProducer, testlogger.New(t))
+			visibilityStore := mgr.(*pinotVisibilityStore)
+
+			mockPinotClient.EXPECT().GetTableName().Return(testTableName).Times(1)
+			mockPinotClient.EXPECT().Search(gomock.Any()).Return(nil, nil).Times(1)
+			resp, err := visibilityStore.ListClosedWorkflowExecutionsByType(context.Background(), test.request)
+			assert.Equal(t, test.expectedResp, resp)
+			assert.Equal(t, test.expectedError, err)
+		})
+	}
+}
+
+func TestListOpenWorkflowExecutionsByWorkflowID(t *testing.T) {
+	request := &p.InternalListWorkflowExecutionsByWorkflowIDRequest{}
+
+	tests := map[string]struct {
+		request       *p.InternalListWorkflowExecutionsByWorkflowIDRequest
+		expectedResp  *p.InternalListWorkflowExecutionsResponse
+		expectedError error
+	}{
+		"Case1: normal case with nil response": {
+			request:       request,
+			expectedResp:  nil,
+			expectedError: nil,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockPinotClient := pnt.NewMockGenericClient(ctrl)
+			mockProducer := &mocks.KafkaProducer{}
+			mgr := NewPinotVisibilityStore(mockPinotClient, &service.Config{
+				ValidSearchAttributes:  dynamicconfig.GetMapPropertyFn(definition.GetDefaultIndexedKeys()),
+				ESIndexMaxResultWindow: dynamicconfig.GetIntPropertyFn(3),
+			}, mockProducer, testlogger.New(t))
+			visibilityStore := mgr.(*pinotVisibilityStore)
+
+			mockPinotClient.EXPECT().GetTableName().Return(testTableName).Times(1)
+			mockPinotClient.EXPECT().Search(gomock.Any()).Return(nil, nil).Times(1)
+			resp, err := visibilityStore.ListOpenWorkflowExecutionsByWorkflowID(context.Background(), test.request)
+			assert.Equal(t, test.expectedResp, resp)
+			assert.Equal(t, test.expectedError, err)
+		})
+	}
+}
+
+func TestListClosedWorkflowExecutionsByWorkflowID(t *testing.T) {
+	request := &p.InternalListWorkflowExecutionsByWorkflowIDRequest{}
+
+	tests := map[string]struct {
+		request       *p.InternalListWorkflowExecutionsByWorkflowIDRequest
+		expectedResp  *p.InternalListWorkflowExecutionsResponse
+		expectedError error
+	}{
+		"Case1: normal case with nil response": {
```

Comment:
Let's cover the error cases as well. maybe in follow up PR you can add more cases to the tests in this file.